### PR TITLE
Introduce GRPC_TRACE_FLAG_ENABLED macro to mark trace branches unlikely.

### PR DESCRIPTION
**Pull request by soheilhy**
This is a trivial change and results in better code generation,
because it marks the trace path unlikely.

The changes are all mechanical, except I did some minor edits in
two macros in call_stack.h and http to apply best practices.

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/18986)

### CODE REVIEW

**Comment by vjpai** [comment]
`do ... while(0)` is the way to make a multi-line multi-statement macro look like a single statement so that you can end its invocation with a semicolon

File: src/core/lib/surface/call.h [hunk 1]
```diff
@@ -102,7 +102,8 @@ void grpc_call_context_set(grpc_call* call, grpc_context_index elem,
 void* grpc_call_context_get(grpc_call* call, grpc_context_index elem);
 
 #define GRPC_CALL_LOG_BATCH(sev, call, ops, nops, tag) \
-  if (grpc_api_trace.enabled()) grpc_call_log_batch(sev, call, ops, nops, tag)
+  if (GRPC_TRACE_FLAG_ENABLED(grpc_api_trace))         \
+  grpc_call_log_batch(sev, call, ops, nops, tag)
```
Comment (vjpai) [inline src/core/lib/surface/call.h]: In the spirit of avoiding programming errors, perhaps this should also be wrapped in something?

File: src/core/lib/surface/completion_queue.cc [hunk 2]
```diff
@@ -411,12 +411,13 @@ static const cq_vtable g_cq_vtable[] = {
...
-    char* _ev = grpc_event_string(event);                                  \
-    gpr_log(GPR_INFO, "RETURN_EVENT[%p]: %s", cq, _ev);                    \
-    gpr_free(_ev);                                                         \
+#define GRPC_SURFACE_TRACE_RETURNED_EVENT(cq, event)    \
+  if (GRPC_TRACE_FLAG_ENABLED(grpc_api_trace) &&        \
+      (GRPC_TRACE_FLAG_ENABLED(grpc_cq_pluck_trace) ||  \
+       (event)->type != GRPC_QUEUE_TIMEOUT)) {          \
+    char* _ev = grpc_event_string(event);               \
+    gpr_log(GPR_INFO, "RETURN_EVENT[%p]: %s", cq, _ev); \
+    gpr_free(_ev);                                      \
```
Comment (vjpai) [inline src/core/lib/surface/completion_queue.cc]: ... and likewise with this?

File: src/core/lib/surface/call.h (hunk 1 above)
Comment (soheilhy) [reply src/core/lib/surface/call.h]: Thank you for noting this. Sure converted this and 4 others over.

File: src/core/lib/surface/completion_queue.cc (hunk 2 above)
Comment (soheilhy) [reply src/core/lib/surface/completion_queue.cc]: Thank you for noting this. Sure converted this and 4 others over.

**Comment by soheilhy** [comment]
Thank you both for approval!

**Comment by soheilhy** [comment]
Weirdly all inter-op tests started failing, they don't look related but I wouldn't submit. I'll force rerun, and will try to understand what's happening (FWIW, interop tests started failing since Monday).

**Comment by soheilhy** [comment]
These are all tool failures :-/ Happening on quite a few different PRs
```
because only the committed changes in the current branch will be
copied to the docker environment or into subworkspaces.
Looking for irrelevant tests to skip...
Finding file differences between gRPC origin/null branch and pull request...

Traceback (most recent call last):
  File "tools/run_tests/run_tests_matrix.py", line 564, in <module>
    relevant_jobs = filter_tests(jobs, args.base_branch)
  File "/Volumes/BuildData/tmpfs/src/github/grpc/tools/run_tests/python_utils/filter_pull_request_tests.py", line 184, in filter_tests
    changed_files = _get_changed_files(base_branch)
  File "/Volumes/BuildData/tmpfs/src/github/grpc/tools/run_tests/python_utils/filter_pull_request_tests.py", line 128, in _get_changed_files
    "HEAD"]).rstrip()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 223, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['git', 'merge-base', 'origin/null', 'HEAD']' returned non-zero exit status 128
```

**Comment by soheilhy** [comment]
All tool failures.

**Comment by soheilhy** [comment]
Thank you everyone for the approvals!

**Pull request merged**